src/hooks/helpers.ts

```diff
@@ -0,0 +1,51 @@
+import { useReducer } from "react";
+
+export interface TransactionState {
+  status: TransactionStateStatus;
+  error: any;
+}
+
+export interface UseEthereumTransactionState {
+  state: TransactionState;
+  setLoading: () => void;
+  setNoWallet: () => void;
+  setReady: () => void;
+  setSuccess: () => void;
+  setMining: () => void;
+  setError: (e: any) => void;
+}
+
+export enum TransactionStateStatus {
```

Comment:
only 3 states are used, what's the point of adding the unused state for the moment ?

Reply:
for for the next hooks Ill need them.